### ISSUE TITLE
ID is inserted as zero

### ISSUE THREAD

**Issue by jainankita90** [rails/rails#47792]
### Steps to reproduce
Create Migration As

```
# frozen_string_literal: true

class CreateAppUsers < ActiveRecord::Migration[7.0]
  def change
  create_table :app_users, id: false do |t|
      ## Database authenticatable
      t.bigint :id, primary_key: true, null: false, default: -> { "(uuid_short())" }
      t.string :email,              null: false, default: ""
      t.string :encrypted_password, null: false, default: ""
      t.timestamps null: false
    end

    add_index :app_users, :email,                unique: true

  end
end
```
Create Model for app_users table
```
class AppUser < ApplicationRecord
  after_create :show_user_id
  def show_user_id
    puts self.id
  end

end
```

### Run Script

Go to rails console
```
AppUser.create(email: 'email@new.com', encrypted_password: 'password')

```

### Expected behavior
#<AppUser:0x00007fad5631dd40
 id: 100243425718173704,
 email: "email@new.com",
 encrypted_password: "[FILTERED]",
 created_at: Tue, 28 Mar 2023 15:50:59.216980000 UTC +00:00,
 updated_at: Tue, 28 Mar 2023 15:50:59.216980000 UTC +00:00>

### Actual behavior
#<AppUser:0x00007fad5631dd40
 id: **0,**
 email: "email@new.com",
 encrypted_password: "[FILTERED]",
 created_at: Tue, 28 Mar 2023 15:50:59.216980000 UTC +00:00,
 updated_at: Tue, 28 Mar 2023 15:50:59.216980000 UTC +00:00>

### System configuration
**Rails version**: Rails 7.0.4.2  

**Ruby version**:
ruby 3.2.1 (2023-02-08 revision 31819e82c8) [x86_64-linux-musl]

**Comment by fatkodima** [comment]
Are you using MySQL?

**Comment by fatkodima** [comment]
Seems like uuid (for ids at least) are not supported in rails for mysql, because mysql does not support `RETURNING` clause (which is used by postgres, for example, to get the id generated by db).

One of the solutions is to remove the default from the migration, but generate the uuid manually in the `before_create` callback in the model.

**Comment by jainankita90** [comment]
yes that helps, but with this I have to add after_create in each model

**Comment by fatkodima** [comment]
You can add that to `ApplicationRecord`, if all the models are using uuids.

**Comment by fatkodima** [comment]
This is an issue with any format of the primary key, where the value is generated by the MySQL.

This is a problem with non-primary keys to, but at least if you have a primary key, you can call `.reload` on the record, and rails will get all the values from the db using the primary key.
